dhis-2/dhis-api/src/main/java/org/hisp/dhis/icon/DefaultIcon.java

```diff
@@ -29,520 +29,712 @@
 
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Date;
 import lombok.Getter;
-import lombok.RequiredArgsConstructor;
 
 /** Default icons are pre-installed immutable icons. */
 @Getter
-@RequiredArgsConstructor
 public class DefaultIcon implements Icon {
   private final String key;
 
   private final String description;
 
   private final String[] keywords;
 
+  private final boolean custom;
+
+  private Date created;
+
+  private Date lastUpdated;
+
+  public DefaultIcon(String key, String description, String[] keywords, boolean custom) {
+    this.key = key;
+    this.description = description;
+    this.keywords = keywords;
+    this.custom = custom;
```

Comment:
similarly to the comment in `customIcon`, I think we custom should always return false in this case.
the way we have it now we could have an instance of `defaultIcon` with the custom flag set to true.